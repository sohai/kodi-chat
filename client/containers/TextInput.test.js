/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import createStore from '../store';

import {
  TextInputRender,
  mapStateToProps,
  mapDispatchToProps
} from './TextInput';

const store = createStore();

describe('(Container) TextInput', () => {
  const state = store.getState();
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });

  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(<TextInputRender {...props} />, {
      context: { store }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
