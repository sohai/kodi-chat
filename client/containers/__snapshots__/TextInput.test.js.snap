// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) TextInput should render 1`] = `<TextInput />`;

exports[`(Container) TextInput will receive right actions 1`] = `
Object {
  "addAndSendMessage": [Function],
  "addMessage": [Function],
  "removeMyLastMessage": [Function],
  "setName": [Function],
  "setTyping": [Function],
}
`;

exports[`(Container) TextInput will receive right props 1`] = `
Object {
  "username": "",
}
`;
